src/components/elementView/elementTab.tsx

```diff
@@ -549,17 +553,9 @@ class ElementTab extends React.Component<ElementTabProps, ElementViewState> {
                 this.props.element !== ElementType.ASSET &&
                 this.props.changePlan
                   ? true
-                  : !(
-                      this.props.permissionState.admin ||
-                      (this.props.element === ElementType.DATACENTER &&
-                        this.props.permissionState.asset_management) ||
-                      (this.props.element === ElementType.MODEL &&
-                        this.props.permissionState.model_management) ||
-                      (this.props.element === ElementType.ASSET &&
-                        this.props.permissionState.asset_management) ||
-                      (this.props.element === ElementType.ASSET &&
-                        this.props.permissionState.datacenter_permissions
-                          .length > 0)
+                  : !hasAddElementPermission(
+                      this.props.element,
+                      this.props.permissionState
```

Comment:
refactor #2 add button

Reply:
Is this function only for add/bulk upload, not edit/decommission/delete because the latter three require checking datacenter? I don't love that we factored this out but not that, but it isn't a huge deal

Reply:
I only did add/bulk upload bc i only refactored elementTab, i can do a full refactor later

Reply:
ok i lied i will do this refactor now